go-controller/pkg/ovn/gress_policy.go

```diff
@@ -0,0 +1,193 @@
+package ovn
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	v1 "k8s.io/api/core/v1"
+	knet "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type gressPolicy struct {
+	namespace  string
+	name       string
+	policyType knet.PolicyType
+	idx        int
+
+	// peerAddressSet points to the addressSet that holds all peer pod
+	// IP addresess.
+	peerAddressSet *addressSet
+
+	// nsAddressSets holds the names of all namespace address sets
+	nsAddressSets sets.String
+
+	// sortedPeerAddressSets has the sorted peerAddressSets
```

Comment:
why do we store sorted peer address sets? If there is a good reason to keep it (which I'm not currently seeing) then why do we store both peer address set and stored peer address set? Why not just store 1 or the other?

Reply:
yeah, I actually fixed this in my address set rewrite: https://github.com/danwinship/ovn-kubernetes/commit/5547fd9b. Feel free to take that.

Reply:
@trozet so that we don't have to sort them every time we use them to create an ACL. More efficient to sort them when adding/removing items from the set.

The peerAddressSet is the set of all pods that this gress policy will allow based on the namespaceSelector+podSelector or just podSelector. There will only be one of these, but the gress policy owns it.

The nsAddressSets are all the address sets selected by a namespaceSelector, of which there may be multiple and which are updated by the namespace watch for the gress policy.

We want to add all of these address sets to the L3 match for the ACL. If you don't sort them then you don't have a stable order to use when finding the address set in handlePeerNamespaceSelectorModify() and other functions. 

This code hasn't changed from before (it's really just moving code from policy.go -> gress_policy.go) so I'd prefer to simplify it later.